### PR TITLE
Port 1.12 Turn of Millking

### PR DESCRIPTION
**Pull request by Asherslab**
# Changes proposed in this pull request:
- Allows users to turn Of and On cowboy milking of cows

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingCowboy.java [hunk 1]
```diff
@@ -67,11 +76,48 @@ public AbstractJob createJob(final CitizenData citizen)
...
+    public void setMilkCows(final boolean milkCows)
+    {
+        this.milkCows = milkCows;
+        markDirty();
+    }
+
     /**
      * ClientSide representation of the building.
      */
     public static class View extends AbstractBuildingWorker.View
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingCowboy.java:114]: MAJOR Reduce this class from 41 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 50 issues
* ![MAJOR][MAJOR] 49 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/f2a87195a1ef4c1373dbef9d7faf8e639a8daf76/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 62 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 57 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L961](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L961): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): class "ColonyManager" has 57 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L330](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L330): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L595](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L595): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [ColonyManager.java#L933](https://github.com/Minecolonies/minecolonies/blob/51577efcf52e96d966aec6761df4b4f7044930e4/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L933): Refactor this method to reduce its Cognitive Complexity from 31 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 62 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 57 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L961](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L961): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): class "ColonyManager" has 57 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L330](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L330): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L595](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L595): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [ColonyManager.java#L933](https://github.com/Minecolonies/minecolonies/blob/18f299c60248613cff84eba4515303dc8db2ad8b/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L933): Refactor this method to reduce its Cognitive Complexity from 31 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**